src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -1711,4 +1759,40 @@ namespace smt::noodler {
         return true;
     }
 
+    std::string FormulaPreprocessor::print_info(bool print_nfas) {
+        std::stringstream res;
+        res << "Current formula:\n";
+        for (const auto& pred : formula.get_predicates_set()) {
+            res << pred << std::endl;
+        }
+        res << "Current automata assignment:\n";
+        for (const auto& [var, nfa] : aut_ass) {
+            res << var << " -> ";
+            if (print_nfas) {
+                res << std::endl << *nfa;
+            } else {
+                res << "NFA\n";
+            }
+        }
+        res << "Current substition map:\n";
+        for (const auto& [var, subst] : substitution_map) {
+            res << var << " ->";
+            for (const auto& subst_var : subst) {
+                res << " " << subst_var;
+            }
+            res << std::endl;
+        }
+        res << "Current removed equations:\n";
+        for (const auto& rem_eq : removed_equations) {
```

Comment:
Can you rename the `removed_equations`? It sounds a bit strange. For example to `model_inclusions`? Or something similar making the relation to models? Because other rules are removing equations, but you don't store them since they are stored in substitution map.

Reply:
I changed it to `removed_inclusions_for_model`